src/tls.c

```diff
@@ -5351,6 +5351,165 @@ int TLSX_EncryptThenMac_Respond(WOLFSSL* ssl)
 
 #endif /* HAVE_ENCRYPT_THEN_MAC && !WOLFSSL_AEAD_ONLY */
 
+
+#if defined(WOLFSSL_DTLS) && defined(WOLFSSL_SRTP)
+
+/******************************************************************************/
+/* DTLS SRTP (Secure Real-time Transport Protocol)                            */
+/******************************************************************************/
+
+/* Only support single SRTP profile */
+typedef struct TlsxSrtp {
+    word16 len;
+    word16 id;
+} TlsxSrtp;
+
+static int TLSX_UseSRTP_GetSize(TlsxSrtp *srtp)
+{
+    /* only supports single profile */
+    const int profileCount = 1;
+    (void)srtp;
+    /*   SRTP Profile Len (2)
+     *      SRTP Profiles (2)
+     *   MKI (master key id) Length */
+    return (OPAQUE16_LEN + (profileCount * OPAQUE16_LEN) + 1);
+}
+
+static TlsxSrtp* TLSX_UseSRTP_New(word16 id, void* heap)
+{
+    TlsxSrtp* srtp;
+
+    srtp = (TlsxSrtp*)XMALLOC(sizeof(TlsxSrtp), heap, DYNAMIC_TYPE_TLSX);
+    if (srtp == NULL) {
+        WOLFSSL_MSG("TLSX SRTP Memory failure");
+        return NULL;
+    }
+    srtp->len = 2;
+    srtp->id = id;
+
+    return srtp;
+}
+
+static void TLSX_UseSRTP_Free(TlsxSrtp *srtp, void* heap)
+{
+    if (srtp != NULL) {
+        XFREE(srtp, heap, DYNAMIC_TYPE_TLSX);
+    }
+    (void)heap;
+}
+
+static int TLSX_UseSRTP_Parse(WOLFSSL* ssl, const byte* input, word16 length,
+    byte isRequest)
+{
+    int ret = BAD_FUNC_ARG;
+#ifndef NO_WOLFSSL_SERVER
+    int i;
+    TlsxSrtp* srtp = NULL;
+    word16 profile_len = 0;
+    word16 profile_value = 0;
+    word16 offset = 0;
+#endif
+
+    if (length < OPAQUE16_LEN) {
+        return BUFFER_ERROR;
+    }
+
+    if (!isRequest) {
+#ifndef NO_WOLFSSL_CLIENT
+        ssl->dtlsSrtpProfile = ssl->ctx->dtlsSrtpProfile;
```

Comment:
Here we should set `ssl->dtlsSrtpId` with the one selected by the server (after checking that is in the list we sent to him)
I used this and worked:
```
modified   src/tls.c
@@ -5421,7 +5421,20 @@ static int TLSX_UseSRTP_Parse(WOLFSSL* ssl, const byte* input, word16 length,
 
     if (!isRequest) {
 #ifndef NO_WOLFSSL_CLIENT
-        ssl->dtlsSrtpProfiles = ssl->ctx->dtlsSrtpProfiles;
+        /* skip length */
+        offset += OPAQUE16_LEN;
+        if (length - offset < OPAQUE16_LEN)
+            return BUFFER_ERROR;
+
+        ato16(input + offset, &profile_value);
+
+        /* check that the profile received was in the ones we support */
+        if ((ssl->dtlsSrtpProfiles & (1 << profile_value)) == 0) {
+            WOLFSSL_MSG("SRP profile sent by the server not supported!");
+            return 0;
+        }
+
+        ssl->dtlsSrtpId = profile_value;
         ret = 0; /* success */
 #endif
     }
```